lattice-controller/src/main.rs

```diff
@@ -0,0 +1,406 @@
+                .map_err(|e| RpcError::InvalidParameter(format!("corrupt config_json: {}", e)))?
+        } else {
+            ConnectionConfig::default()
+        };
+
+        if let Some(url) = values.get(ENV_NATS_URI) {
+            config.cluster_uris.push(url.clone());
+        }
+        if let Some(jwt) = values.get(ENV_NATS_CLIENT_JWT) {
+            config.auth_jwt = Some(jwt.clone());
+        }
+        if let Some(seed) = values.get(ENV_NATS_CLIENT_SEED) {
+            config.auth_seed = Some(seed.clone());
+        }
+        if config.auth_jwt.is_some() && config.auth_seed.is_none() {
+            return Err(RpcError::InvalidParameter(
+                "if you specify jwt, you must also specify a seed".to_string(),
+            ));
+        }
+        if config.cluster_uris.is_empty() {
+            config.cluster_uris.push(DEFAULT_NATS_URI.to_string());
+        }
+
+        if let Some(nsprefix) = values.get(ENV_LATTICE_PREFIX) {
+            config.lattice_prefix = nsprefix.to_owned();
+        }
+
+        if let Some(auction_timeout) = values.get(ENV_AUCTION_TIMEOUT_MS) {
+            config.auction_timeout_ms = auction_timeout.parse().unwrap_or(3 * DEFAULT_TIMEOUT_MS)
+        }
+
+        if let Some(timeout) = values.get(ENV_TIMEOUT_MS) {
+            config.timeout_ms = timeout.parse().unwrap_or(DEFAULT_TIMEOUT_MS);
+        }
+
+        if config.timeout_ms == 0 {
+            config.timeout_ms = DEFAULT_TIMEOUT_MS
+        }
+
+        if config.auction_timeout_ms == 0 {
+            config.auction_timeout_ms = 3 * DEFAULT_TIMEOUT_MS
+        }
+
+        if config.lattice_prefix.is_empty() {
+            config.lattice_prefix = "default".to_owned()
+        }
+
+        Ok(config)
+    }
+}
+
+// main (via provider_main) initializes the threaded tokio executor,
+// listens to lattice rpcs, handles actor links,
+// and returns only when it receives a shutdown message
+//
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    provider_main(LatticeControllerProvider::default())?;
+
+    eprintln!("Lattice Controller capability provider exiting");
+    Ok(())
+}
+
+/// lattice-controller capability provider implementation
+#[derive(Default, Clone, Provider)]
+#[services(LatticeController)]
+struct LatticeControllerProvider {
+    configs: Arc<RwLock<HashMap<String, ConnectionConfig>>>,
+}
+
+impl LatticeControllerProvider {
+    async fn connect(&self, cfg: ConnectionConfig) -> Result<Connection, RpcError> {
+        let mut opts = match (cfg.auth_jwt, cfg.auth_seed) {
+            (Some(jwt), Some(seed)) => {
+                let kp = KeyPair::from_seed(&seed)
+                    .map_err(|e| RpcError::ProviderInit(format!("key init: {}", e)))?;
+                asynk::Options::with_jwt(
+                    move || Ok(jwt.clone()),
+                    move |nonce| kp.sign(nonce).unwrap(),
+                )
+            }
+            (None, None) => asynk::Options::new(),
+            _ => {
```

Comment:
Note: It is possible to do _just_ seed authentication without a JWT and I just added that to `wash` as a part of the context PR (mostly because it enabled my demo of seed auth). Not saying we have to add that quite yet if that isn't an option we want to support, but that is a valid authentication scheme

Reply:
right. this is the same set of authentication options the NATS provider supports. we can always upgrade later.